src/app/public/modules/dynamic-component/dynamic-component.service.ts

```diff
@@ -0,0 +1,80 @@
+//#region imports
+
+import {
+  ApplicationRef,
+  ComponentFactoryResolver,
+  ComponentRef,
+  EmbeddedViewRef,
+  Injectable,
+  Injector,
+  RendererFactory2,
+  Renderer2
+} from '@angular/core';
+
+import {
+  SkyWindowRefService
+} from '../window';
+
+import {
+  SkyDynamicComponentArgs
+} from './dynamic-component-args';
+
+import {
+  SkyDynamicComponentLocation
+} from './dynamic-component-location';
+
+//#endregion
+
+/**
+ * Angular service for creating and rendering a dynamic component.
+ */
+@Injectable()
+export class SkyDynamicComponentService {
+
+  private renderer: Renderer2;
+
+  constructor(
+    private componentFactoryResolver: ComponentFactoryResolver,
+    private appRef: ApplicationRef,
+    private injector: Injector,
+    private windowRef: SkyWindowRefService,
+    rendererFactory: RendererFactory2
+  ) {
+    // Based on suggestions from https://github.com/angular/angular/issues/17824
+    // for accessing an instance of Renderer2 in a service since Renderer2 can't
+    // be injected into a service.  Passing undefined for both parameters results
+    // in the default renderer which is what we want here.
+    this.renderer = rendererFactory.createRenderer(undefined, undefined);
+  }
+
+  /**
+   * Creates an instance of the specified component and adds it to the specified location
+   * on the page.
+   * @param args Options for creating the dynamic component.
+   */
+  public createComponent<T>(args: SkyDynamicComponentArgs): ComponentRef<T> {
```

Comment:
When we're injecting dynamic components in our other libraries (flyout and toast, etc.) we've created the pattern of having the component type be the first parameter, and the options be the second. See:
https://github.com/blackbaud/skyux-flyout/blob/master/src/app/public/modules/flyout/flyout.service.ts#L60
https://github.com/blackbaud/skyux-toast/blob/master/src/app/public/modules/toast/toast.service.ts#L100

This makes it easier to apply the component's type through `Type<T>` and have the type returned to the consumer (currently, the returned component reference does not know about the injected component's type). I created a custom component to test this out and the IDE didn't pick up my component's API:

<img width="639" alt="screen shot 2018-11-28 at 5 28 30 pm" src="https://user-images.githubusercontent.com/12497062/49186610-2259eb00-f333-11e8-9cc3-ce6ae698801b.png">


To get the type to come through, we can do this:

**dynamic-component.service.ts
(I personally like this better than the second option)**
```
// Allow for configuration defaults.
public createComponent<T>(
  component: Type<T>,
  args?: SkyDynamicComponentArgs
): ComponentRef<T>
```

If you don't like that, we'll need to pass in the component's type into the args interface so that the IDE recognizes the `componentRef.instance` members.

**dynamic-component.service.ts**
```
public createComponent<T>(
  args: SkyDynamicComponentArgs<Type<T>>
): ComponentRef<T> { }
```

**dynamic-component-args.ts**
```
export interface SkyDynamicComponentArgs<T> {

  componentType: T;

  // ...
}
```

Reply:
I made the suggested change where `componentType` is now its own parameter.